### ISSUE TITLE
Feature request: solvent accessible surface area (SASA)

### ISSUE THREAD

**Issue by f3rmion** [AstraZeneca/kallisto#10]
**Is your feature request related to a problem? Please describe.**
Implement solvent accessible surface area (SASA) using a radial Lebedev-Laikov grid as described in "A quadrature formula for the sphere of the 131st algebraic order of accuracy", Doklady Mathematics, Vol. 59, No. 3, 1999, pp. 477-481.

**Describe the solution you'd like**
Extend present grid implementation in ``src/grid.py`` to cover finer grids and use the defined gridpoints to construct a SASA.

**Comment by awvwgk** [comment]
Note that most Lebedev-Laikov grid implementations carry either the LGPL (inherited from [J. Burkardt's version](https://people.sc.fsu.edu/~jburkardt/datasets/sphere_lebedev_rule/sphere_lebedev_rule.html)) or originate from the CCL announcement of the [original implementation of D. Laikov](http://www.ccl.net/cca/software/SOURCES/FORTRAN/Lebedev-Laikov-Grids/), which has somewhat unclear licensing conditions but might be considered public domain (?).

**Comment by f3rmion** [comment]
> Note that most Lebedev-Laikov grid implementations carry either the LGPL (inherited from [J. Burkardt's version](https://people.sc.fsu.edu/~jburkardt/datasets/sphere_lebedev_rule/sphere_lebedev_rule.html)) or originate from the CCL announcement of the [original implementation of D. Laikov](http://www.ccl.net/cca/software/SOURCES/FORTRAN/Lebedev-Laikov-Grids/), which has somewhat unclear licensing conditions but might be considered public domain (?).

That's right, using Burkardt's version would force an LGPL license on the project. Laikov's original version does not seem to be properly licensed, especially since the linked Fortran version was not implemented by Laikov himself, but rather by Christoph van Wuellen depending on an initial C-version written by Laikov.

**Comment by awvwgk** [comment]
My [`xtb`](https://github.com/grimme-lab/xtb/blob/master/src/solv/lebedev.f90) and [DFTB+](https://github.com/dftbplus/dftbplus/blob/master/prog/dftb%2B/lib_extlibs/lebedev.F90) implementation of the SASA is based on J. Burkhardt's version due to the clearer and compatible licensing.

Every other implementation ([C](https://github.com/Rufflewind/lebedev_laikov), [Matlab](https://www.mathworks.com/matlabcentral/fileexchange/27097-getlebedevsphere), [Python](https://github.com/nschloe/quadpy), ...) I have been able to find so far at some point goes back to J. Burkhardt's version or C. van Wuellen's version of the C implementation by D. Laikov. The original C code seems to be lost, at least I wasn't able to find it up to now.

**Comment by Montana** [comment]
Thank you @awvwgk!
